src/datablocks/datablock.go

```diff
@@ -17,26 +17,39 @@ type DataBlock struct {
 	values    []*DataBlockValue
 	indexmap  map[string]int
 	valuesmap map[string]*DataBlockValue
+
+	// start, end for block slice
+	start     int
```

Comment:
block.start and end are very very hard to maintain, since in datablocks, other method(such as datablock_filter/orderby) maybe deal with blocks.values directly, contants variable doesn't work any more.

And for limit, I think delete/re-slice block.seqs slice is ok:
limit 5:
blocks.seqs = blocks.seqs[:5]
or limit 1, 5
blocks.seqs = blocks.seqs[1:5]

Throuth, we can iterate over the row/column-iterator for the right results.